### PR TITLE
Support nested fields

### PR DESCRIPTION
**Pull request by yesmeck**
I didn't touch any validation related codes. Validation can be wrote as before.

Example: 

https://github.com/yesmeck/redux-form/blob/gh-pages/src/examples/HighOrderNestedForm.js
https://github.com/yesmeck/redux-form/blob/gh-pages/src/examples/NestedForm.js

### CODE REVIEW

File: src/readFields.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import isPristine from './isPristine';
 import isValid from './isValid';
 import getValues from './getValues';
 import silencePromise from './silencePromise';
+import {get} from 'lodash';
```
Comment (vincentfretin) [inline src/readFields.js]: If you want to reduce the size of the generated bundle, use:
import get from 'lodash/object/get';

Comment (yesmeck) [inline src/readFields.js]: Thanks, fixed.

**Comment by yesmeck** [comment]
@erikras A little thing, the action `initialize` need a extra argument `fields now, it breaks the compatibility.

**Comment by rogeriochaves** [comment]
@yesmeck @erikras any updates on this? It looks great, I'm really needing it

**Comment by yesmeck** [comment]
As @erikras said he will [implement](https://github.com/erikras/redux-form/issues/55#issuecomment-154722417)  something on this, I'm closing this hacky implementing.

**Pull request closed without merging**